ls1intum/Artemis
# `User management`: Add filter for registration number

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


This change allows the admin user to filter (besides the existing ones) by registration number and to sync the user info with the LDAP server.

### Description

I added a new filter option to the filter part of the user management and a new button that allows the admin to sync with the the LDAP server. 

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to User Management
3. On the main page click on the LDAP button for users with registration number (it should be available only for the users with the LDAP profile active. It is not possible to test on the tests servers since they do not have a LDAP connection).
4. Open the filter option and filter users with registration number, without registration number and with none filter.

### Review Progress





#### Code Review

#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots


![image](https://user-images.githubusercontent.com/64961233/176622427-9c9d6814-5d2c-48f9-9696-9088fe17df39.png)
![image](https://user-images.githubusercontent.com/64961233/176622502-8147eec3-2955-4fd3-af4e-10adbc848f51.png)
